users/utils/open_city_profile/birthday_resolver.py

```diff
@@ -16,10 +19,14 @@
 def resolve_user(request, payload):
     user = get_or_create_user(payload, oidc=True)
 
-    if not user.profile_id:
-        profile_id_reader = ProfileNodeIdReader(request)
-        user.profile_id = profile_id_reader.get_user_profile_id()
-        user.save()
+    if not user.profile_id and not user.has_staff_permissions:
+        try:
+            profile_id_reader = ProfileNodeIdReader(request)
+            user.profile_id = profile_id_reader.get_user_profile_id()
+        except ProfileReadError as e:
+            capture_exception(e)
+        else:
+            user.save()
```

Comment:
I did not know one can use `else` with `try`!

I'm struggling a bit to understand the benefit, but I guess you did it like this because you want `try` block to be about profile reading and not about saving the user. So any errors thrown from `user.save()` and not handled. Or did I miss something?

Reply:
Yes exactly. Try to read the profile id and if success save it. Of course it could also be inside the try block but I like it like this.